src/app/components/analytics/host-analytics/host-analytics.component.spec.ts

```diff
@@ -47,12 +49,102 @@ describe('HostAnalyticsComponent', () => {
   }));
 
   beforeEach(() => {
+    apiService = TestBed.get(ApiService);
     fixture = TestBed.createComponent(HostAnalyticsComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
+    fixture.detectChanges();
+    const err = { 'status': 403, 'error': '' };
+    component.errCallBack(err);
+    err['status'] = 401;
+    component.errCallBack(err);
     expect(component).toBeTruthy();
   });
+
+  it('should call getHostTeam', () => {
+    fixture.detectChanges();
+    spyOn(apiService, 'getUrl').and.returnValue(new Observable((observer) => {
+      observer.next({'status': 200});
+      observer.complete();
+      return {unsubscribe() {}};
+    }));
+    fixture.detectChanges();
+    component.getHostTeam();
+    expect(apiService.getUrl).toHaveBeenCalled();
+  });
+
+  it('should give error for getHostTeam', () => {
+    fixture.detectChanges();
+    const err = { 'status': 403, 'error': '' };
+    spyOn(apiService, 'getUrl').and.returnValue(new Observable((observer) => {
+      observer.error(err);
+      return {unsubscribe() {}};
+    }));
+    fixture.detectChanges();
+    component.getHostTeam();
+
+    err['status'] = 401;
+    component.getHostTeam();
+    expect(apiService.getUrl).toHaveBeenCalled();
```

Comment:
In these specs, should we also check if `globalService.showToast` is called with the desired parameters?

Reply:
Not required as showToast method is specific to globalService but not to these component specs.